cub/cub/util_type.cuh

```diff
@@ -411,29 +394,26 @@ struct AlignBytes
     char byte;
   };
 
-  enum
-  {
-    /// The "true CUDA" alignment of T in bytes
-    ALIGN_BYTES = sizeof(Pad) - sizeof(T)
-  };
+  /// The "true CUDA" alignment of T in bytes
+  static constexpr unsigned ALIGN_BYTES = sizeof(Pad) - sizeof(T);
+
+  static_assert(ALIGN_BYTES == alignof(T), "WTF is going on?");
```

Comment:
:D Sorry about that, I should have added a comment to explain my intention. I wanted to push this `static_assert` into the CI to see whether it fails for any compiler. This line was never intended to be merged into the production code base.

I went for using `alignof(T)` directly, since it works for all compilers tested by the CI.